python/ml4ir/applications/ranking/tests/test_feature_fns.py

```
@@ -90,4 +90,37 @@ def test_sequence_categorical_vector_one_hot(self):
                                      domain_2_one_hot,
                                      oov_one_hot,
                                      oov_one_hot])
-        self.assertTrue(np.isclose(actual_one_hot, expected_one_hot).all())
\ No newline at end of file
+        self.assertTrue(np.isclose(actual_one_hot, expected_one_hot).all())
+
+    def test_theoretical_min_max_norm(self):
+        """
+        Test TheoreticalMinMaxNormalization feature transform
+        """
+        feature_info = copy.deepcopy(FEATURE_INFO)
+        feature_info["feature_layer_info"]["args"]["theoretical_min"] = 0.5
+
+        input_feature = np.array([[-0.2, -0.1, 0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0]])
```

should raise an exception

---

or a warning.